src/System.IO.Packaging/src/System/IO/Packaging/ContentType.cs

```diff
@@ -0,0 +1,744 @@
+        /// expanded to optionally compare parameters as well.
+        /// </summary>
+        internal class StrongComparer : IEqualityComparer<ContentType>
+        {
+            /// <summary>
+            /// This method does a strong comparison of the content types.
+            /// Only compares the ContentTypes' type and subtype components.
+            /// </summary>
+            public bool Equals(ContentType x, ContentType y)
+            {
+                if (x == null)
+                {
+                    return (y == null);
+                }
+                else
+                {
+                    return x.AreTypeAndSubTypeEqual(y);
+                }
+            }
+
+            /// <summary>
+            /// We lower case the results of ToString() because it returns the original
+            /// casing passed into the constructor.  ContentTypes that are equal (which
+            /// ignores casing) must have the same hash code.
+            /// </summary>
+            public int GetHashCode(ContentType obj)
+            {
+                return obj.ToString().ToUpperInvariant().GetHashCode();
+            }
+        }
+
+        internal class WeakComparer : IEqualityComparer<ContentType>
+        {
+            /// <summary>            
+            /// This method does a weak comparison of the content types. 
+            /// Parameter and value pairs are not used for the comparison. 
+            /// </summary>
+            public bool Equals(ContentType x, ContentType y)
+            {
+                if (x == null)
+                {
+                    return (y == null);
+                }
+                else
+                {
+                    return x.AreTypeAndSubTypeEqual(y, true);
+                }
+            }
+
+            /// <summary>
+            /// We lower case the results of ToString() because it returns the original
+            /// casing passed into the constructor.  ContentTypes that are equal (which
+            /// ignores casing) must have the same hash code.
+            /// </summary>
+            public int GetHashCode(ContentType obj)
+            {
+                return obj._type.ToUpperInvariant().GetHashCode() ^ obj._subType.ToUpperInvariant().GetHashCode();
+            }
+        }
+        #endregion Nested Classes
+
+        //------------------------------------------------------
+        //
+        //  Private Methods
+        //
+        //------------------------------------------------------
+
+        #region Private Methods
+
+
+        /// <summary>
+        /// This method validates if the content type string has
+        /// valid CR-LF characters. Specifically we test if '\r' is 
+        /// accompanied by a '\n' in the string, else its an error.
+        /// </summary>
+        /// <param name="contentType"></param>
+        private static void ValidateCarriageReturns(string contentType)
```

Comment:
This is a very interesting issue.  RFC-2616 (definition of content types, as referred to by OPC) does not allow for CRLF (or CR or LF) in a content type.  Further, I did a check in my sample documents, and no Open XML documents contain a CR, LF, or CRLF in a content type.  I suspect that this code has never been executed in a real scenario.

However, I am loath to change this code, as changing it could potentially impact existing applications.  There may be some scenario where programs are setting content types with CRLF in them, although I can't think of what that would be.

One of my goals with this module is to implement as closely as possible the existing implementation.  There are certainly aspects of System.IO.Packaging that could be improved upon, such as adding new functionality to make it more LINQ friendly, and so on.  However, I propose that this is a separate project from the project of getting an implementation that is as compatible as possible as the existing one.

I propose that we leave the code as-is for now, and if there are issues that crop up in the Linux or Mac world, we address those as a separate work item.